### DISCUSSION TITLE
Upload progress for a large file

### DISCUSSION THREAD

**Discussion by PushpakBhoge** [fastapi/fastapi#5140] [Questions] [answered]
Hi my project requires uploading of single large binary file (mostly zip) ranging anywhere from 1gb to ~30gb

how can I show upload progress information on  client side?

Also what are the best practices to upload such big files?

PS* -> seperately uploading them to a S3 bucket and then giving path is not an option. Since don't want to keep bucket access on client side

**Comment by Wason1797** [comment, marked as answer]
Maybe you can check [this](https://gist.github.com/virolea/e1af9359fe071f24de3da3500ff0f429) example. I don't really know what tech are you using for your client, but if it is a web application you can track progress in this way.
